File: packages/node/src/headers.ts
import type { X402Fields } from "./aad.js";

export function buildX402Headers(x: X402Fields): Record<string, string> {
  return {
    "X-X402-Invoice-Id": x.invoiceId,
    "X-X402-Chain-Id": String(x.chainId),
    "X-X402-Token-Contract": x.tokenContract,
    "X-X402-Amount": x.amount,
    "X-X402-Recipient": x.recipient,
    "X-X402-Tx-Hash": x.txHash,
    "X-X402-Expiry": String(x.expiry),
    "X-X402-Price-Hash": x.priceHash,
  };
}

File: packages/python/x402_hpke/headers.py
from typing import Dict


def build_x402_headers(x: Dict) -> Dict[str, str]:
    return {
        "X-X402-Invoice-Id": str(x["invoiceId"]),
        "X-X402-Chain-Id": str(int(x["chainId"])),
        "X-X402-Token-Contract": str(x["tokenContract"]),
        "X-X402-Amount": str(x["amount"]),
        "X-X402-Recipient": str(x["recipient"]),
        "X-X402-Tx-Hash": str(x["txHash"]),
        "X-X402-Expiry": str(int(x["expiry"])),
        "X-X402-Price-Hash": str(x["priceHash"]),
    }